src/discordcr/rest.cr

```diff
@@ -2247,5 +2251,515 @@ module Discord
+        application_id,
+        "PUT",
+        "/applications/#{application_id}/guilds/#{guild_id}/commands",
+        HTTP::Headers{"Content-Type" => "application/json"},
+        commands.to_json
+      )
+
+      Array(ApplicationCommand).from_json(response.body)
+    end
+
+    # Fetches command permissions for all commands for your application in a guild.
+    #
+    # [API docs for this method](https://discord.com/developers/docs/interactions/application-commands#get-guild-application-command-permissions)
+    def get_guild_application_command_permissions(guild_id : UInt64 | Snowflake)
+      application_id = client_id
+
+      response = request(
+        :applications_aid_guilds_gid_commands_permissions,
+        application_id,
+        "GET",
+        "/applications/#{application_id}/guilds/#{guild_id}/commands/permissions",
+        HTTP::Headers.new,
+        nil
+      )
+
+      Array(GuildApplicationCommandPermissions).from_json(response.body)
+    end
+
+    # Fetches command permissions for a specific command for your application in a guild.
+    #
+    # [API docs for this method](https://discord.com/developers/docs/interactions/application-commands#get-application-command-permissions)
+    def get_application_command_permissions(guild_id : UInt64 | Snowflake, command_id : UInt64 | Snowflake)
+      application_id = client_id
+
+      response = request(
+        :applications_aid_guilds_gid_commands_cid_permissions,
+        application_id,
+        "GET",
+        "/applications/#{application_id}/guilds/#{guild_id}/commands/#{command_id}/permissions",
+        HTTP::Headers.new,
+        nil
+      )
+
+      GuildApplicationCommandPermissions.from_json(response.body)
+    end
+
+    # Edits command permissions for a specific command for your application in a guild. You can only add up to 10 permission overwrites for a command.
+    #
+    # NOTE: This endpoint requires authentication with a Bearer token that has permission to manage the guild and its roles. For more information, read above about [application command permissions](https://discord.com/developers/docs/interactions/application-commands#permissions).
+    # NOTE: This endpoint will overwrite existing permissions for the command in that guild
+    #
+    # [API docs for this method](https://discord.com/developers/docs/interactions/application-commands#edit-application-command-permissions)
+    def edit_application_command_permissions(guild_id : UInt64 | Snowflake, command_id : UInt64 | Snowflake,
+                                             permissions : Array(ApplicationCommandPermissions), token : String)
+      application_id = client_id
+
+      response = request(
+        :applications_aid_guilds_gid_commands_cid_permissions,
+        application_id,
+        "PUT",
+        "/applications/#{application_id}/guilds/#{guild_id}/commands/#{command_id}/permissions",
+        HTTP::Headers{"Content-Type" => "application/json", "Authorization" => token},
```

Comment:
Could this be automatically prefixed with Bearer?

Reply:
I don't think only Bearer tokens have access to this

Reply:
Only Bearer and user tokens have access

Reply:
Right, I forgot this was broken in v2. I'm still not sure about automatic prefixing because it would be inconsistent with the rest of the lib